### DISCUSSION TITLE
Maintaining Indices

### DISCUSSION THREAD

**Discussion by thatcort** [lance-format/lance#1656] [General]
Hi, how do you update indices when adding rows? I see the methods `create_index()` and `create_scalar_index()`, but it seems like these will become obsolete on every append. Are they automatically updated or is there an update method that needs to be called? Similarly, are the indices updated during table compaction?

**Comment by eddyxu** [comment]
@thatcort we have a method call `optimized_indices()` will append new rows into existing index fragments.

**Reply by king4face** [reply]
> @thatcort we have a method call `optimized_indices()` will append new rows into existing index fragments.

Hi, I'm wondering whether `optimize_indices()` will update the number of partitions of the IVF_PQ index type. If not, is there any way to increase the number of partitions as the dataset grows?

Say we have a dataset with around 5 million rows in the beginning and day by day it will grow to the size of 200 million rows. How do I maintain the index in this scenario? Do I set the partition number according to the final size of the dataset, or start with a small number and update it as the dataset grows?

**Reply by hfutatzhanghb** [reply]
@eddyxu Hi, could you please share how to maintain indices in your practice? Thanks a lot!
